protocol/protocol.typeHierarchy.proposed.md

```diff
@@ -0,0 +1,77 @@
+
+#### Type Hierarchy
```

Comment:
Is there a reason this proposal is using the term "type" everywhere and not "symbol"? For example in TypeScript, those are two very different concepts

Reply:
I think those are very different concepts on most languages.